### ISSUE TITLE
A custom curses script requires enter to be pressed

### ISSUE THREAD

**Issue by seanfarley** [xonsh/xonsh#3273]
It seems that xonsh is overriding the `cbreak` method of `curses` (but I'm not quite sure). Here is a minimal script that demonstrates this behavior on my mac. I've tried this in Terminal iTerm, and Alacritty just to make sure it's not something there.

## xonfig

<details>
```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.9.9                |
| Git SHA          | 5aac8711             |
| Commit Date      | Jul 19 17:43:33 2019 |
| Python           | 3.7.4                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 2.0.9                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.4.2                |
| on posix         | True                 |
| on linux         | False                |
| on darwin        | True                 |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

</details>

## Expected Behavior
Pressing a key should pass it through to `curses` without 1) displaying it on the screen (since `curses.noecho()` is enabled) and 2) waiting (since `curses.cbreak()` is enabled as well)

## Current Behavior
Pressing a key displays the key code on the screen. Nothing happens until `enter` is pressed which is incorrect (it should pass the keypress to the script immediately).

## Steps to Reproduce

Minimal script:

```python
#!/usr/bin/env python

import curses


def main(screen):
    # turn off input echoing
    curses.noecho()

    # respond to keys immediately (don't wait for enter)
    curses.cbreak()

    # map arrow keys to special values
    screen.keypad(True)

    screen.addstr(0, 0, "Press 'q' to quit")

    i = 0
    while True:
        i += 1

        char = screen.getch()

        if char in (curses.KEY_UP, curses.KEY_DOWN, curses.KEY_RIGHT,
                    curses.KEY_LEFT):
            screen.addstr(1, 0, f'Driving {curses.keyname(char).decode():<20} {i}')
        else:
            if char == ord('q'):
                break


curses.wrapper(main)
```

Run it in `bash` to see the difference with `xonsh`.

**Comment by scopatz** [comment]
Curses applications usually need to be run in "unthreadable" mode.  You can add something like the following to your session or xonshrc:

```
 __xonsh__.commands_cache.threadable_predictors['cmd'] = lambda x: False
```

**Comment by seanfarley** [comment]
Sorry for the delay but I finally tested this and it works. Is there any easier way to run curses in this unthreadable mode? e.g. `unthreadable python curses_script.py`?

**Comment by seanfarley** [comment]
(testing a comment)

**Comment by scopatz** [comment]
Hmmm no, but that is a good idea...

**Comment by anki-code** [comment]
Closed with reference to #4214
